portfolio/src/main/webapp/js/script.js

```diff
@@ -12,59 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/********************************************************
- * Listens to the about link in the navigation bar and
- * turns the text to yellow when the about section is visited.  
+/**
+    Fetch demo
  */
-document.getElementById('about-link').addEventListener('click', () => {
-    let about = document.getElementById('about-link')
-    let contact = document.getElementById('contact-link')
-    let projects = document.getElementById('projects-link')
-
-    if(!about.classList.contains('yellow-text'))
-  	    about.classList.add('yellow-text')
-
-    if(contact.classList.contains('yellow-text'))
-  	    contact.classList.remove('yellow-text')
-
-    if(projects.classList.contains('yellow-text'))
-        projects.classList.remove('yellow-text') 
-})
-
-/********************************************************
- * Listens to the projects link in the navigation bar and
- * turns the text to yellow when the projects section is visited.  
- */
-document.getElementById('projects-link').addEventListener('click', () => {
-    let about = document.getElementById('about-link')
-    let contact = document.getElementById('contact-link')
-    let projects = document.getElementById('projects-link')
-
-    if(!projects.classList.contains('yellow-text'))
-        projects.classList.add('yellow-text')
-
-    if(contact.classList.contains('yellow-text'))
-        contact.classList.remove('yellow-text')
-
-    if(about.classList.contains('yellow-text'))
-        about.classList.remove('yellow-text')
-})
-
-/********************************************************
- * Listens to the contact link in the navigation bar and
- * turns the text to yellow when the contact section is visited.  
- */
-document.getElementById('contact-link').addEventListener('click', () => {
-    let about = document.getElementById('about-link')
-    let contact = document.getElementById('contact-link')
-    let projects = document.getElementById('projects-link')
-
-    if(!contact.classList.contains('yellow-text'))
-        contact.classList.add('yellow-text')
-
-    if(about.classList.contains('yellow-text'))
-        about.classList.remove('yellow-text')
-
-    if(projects.classList.contains('yellow-text'))
-        projects.classList.remove('yellow-text')
-})
+ function fetchData(){
+     fetch('/data')
+     .then((response) => response.text()
```

Comment:
Looks like the parentheses don't close out here.  I guess what you may want is `then(response => response.text())`?

Reply:
Hi @qidichen. The I passed two callbacks inside the first then, one that is to be called when the fetch() was successful and the other that is to be called when the fetch() failed.

the second `then` is actually called on `response.text()`. It compiles and works as expected but I think I aligned `then`s clearly made it hard to read. 

Here is what I am trying to do: 

```
fetch('data').then((response) => {
response.text().then((text) => showData(text))
}, () => console.log("Failed to get data at /data"))

```

Reply:
I see, thanks for the clarification.  Generally speaking, deeply nested arrow functions will harm readability and may be error-prone.  You may want to break down it a little bit, but this is just a nit.